feat: show connected devices

itsspriyansh opened the pull request:
### Description
This PR adds functionality for showing a list of currently connected real and emulator devices. The workflow is triggered when a user uses the `connect` subcommand.
* If no option is passed with the `connect` subcommand: show connected real and emulator devices.
* If `--wireless` option is passed: show connected real devices.
* If `--emulator` option is passed: show connected emulators.

<img width="958" alt="image" src="https://github.com/user-attachments/assets/a3363ca7-37ee-4b13-9a1f-12c577e2c6af">

### Relevant discussion: 
- https://discord.com/channels/618399631038218240/1238051267914235904/1238115674954530878
- https://github.com/itsspriyansh/GSoC24/commit/5e3b1577df8564a2f6121c39cb189fae6a77fc4d#commitcomment-142314261

garg3133 commented:
Since `connect` is a "command" here, it does not make much sense to have the `--list` flag show all the devices already connected and not those we can potentially connect to.

The idea here I think was that if we just pass the `connect` subcommand without any corresponding flags, it would show all the devices already connected along with an questionnaire for whether to connect to an AVD or a real device for the next one. But having a `--list` flag just to show the devices we are already connected to does not fit right with the `connect` subcommand. Maybe we can have another `devices` subcommand whose job would be to do just that?

I remember that we had some discussion on this topic previously as well but can't remember the specific details on what we decided at that time. Can you link those discussions as well in this PR? It's always a good idea to link the related discussions in a PR so that no context is lost.

Also, regarding your screenshot above, we should primarily show the device id only for the real devices as well because that's how SDK tools identify those devices when running some commands over them. The IP and port number of the device can be shown in parenthesis maybe at the end of the line?

garg3133 commented:
(To be extra clear, we still want to show the already connected devices, but in the case of `... android connect` and not for `... android connect --list`.)

And for the `--wireless` and `--emulator` flags, we'd first show all the currently connected real devices in case of `--wireless` and all the currently connected AVDs in case of `--emulator` and then proceed with the respective flow for each.

For `--wireless`, it would be to show all the paired devices and allow users to choose from them (as discussed in last meeting) along with an option to pair a new device.

For `--emulator`, it would be to show all the AVDs available to be connected to and allow users to choose from them (with probably an option to create a new AVD as well, but not a priority right now to implement this as well).

itsspriyansh commented:
@garg3133 Thanks for pointing this out. I actually misinterpreted your earlier message on discord (reference attached in PR description). I have updated this PR with the following changes:

I have removed the `--list` flag functionality and moved the code to a new file `subcommands/common.ts`. I have created this new file as this will be used for various common functions like verifying options of subcommands and showing help for individual subcommand.

I am not showing any logs if no device is connected since we aren't using this function to exclusively check the connected devices anymore. If connected devices are found then it will display the list otherwise it won't do anything. Tell me if this works?

itsspriyansh commented:
About the real devices, the \<ip-address:port\> combination serves as the ID for wirelessly connected real devices. So you can use adb just as follows in reference to the above image:

```
adb -s 192.168.29.228:37973 <command>
```

garg3133 commented:
> Tell me if this works?

Yes, that works!

> Or do we need to more dedicated functions.

Yeah, that'd be better.

garg3133 commented on src/commands/android/subcommands/common.ts:
```diff
@@ -0,0 +1,52 @@
+import colors from 'ansi-colors';
+import ADB from 'appium-adb';
+
+import {Options} from '../interfaces';
+import Logger from '../../../logger';
```
I meant the opposite of this. Put outermost relative imports at the top and innermost at the bottom. And for collisions like multiple imports from the same level, follow the alphabetical order.

For example, if we have `../common` and `../utils/adb`, since both have equal numbers of `../`, we would place them in the alphabetical order of the first name after `../`, i.e., comparing `command` and `utils` in alphabetical order.

Let's keep fixing the imports in whichever file we touch.

itsspriyansh commented:
@garg3133 PR updated.

garg3133 commented:
@itsspriyansh Please update the description of this PR to what this PR actually do now.
